hyperspy/io_plugins/hspy.py

```diff
@@ -743,6 +744,16 @@ def write_signal(signal, group, **kwds):
         for model in model_group.values():
             model.attrs['_signal'] = group.name
 
+def get_h5py_read_write_filters():
+    """Get tuple of lists of available read and write compression formats.
+    Some of these, like szip, are dependent on the HDF5 install due to 
+    licensing issues
+    """
+    from h5py._hl.filters import _gen_filter_tuples
```

Comment:
We should not rely on private API of h5py. And if this is used only in the test suite, I don't see why it would be in hyperspy public API.